text/0000-rust-analyzer.md

```diff
@@ -0,0 +1,156 @@
+- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
+- Start Date: 2020-04-20
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+- master [compiler team IDE strategy and planning](/9qrJYdSLTK2MWG4-pbmpWw?edit) document
+
+# Summary
+[summary]: #summary
+
+The RFC proposes a plan to adopt rust-analyzer as Rust's official LSP implementation. The transition to rust-analyzer will take place in a few stages:
+
+* **Feedback** -- encourage people to use rust-analyzer and report problems
+* **Deprecation** -- actively transition people from the RLS to rust-analyzer
+* **Transition** -- stop supporting the older RLS
+
+As detailed below, one major concern with rust-analyzer as it stands today is that it shares very little code with rustc. To avoid creating an unsustainable maintainance burden, this RFC extracting shared libraries that will be used by both rustc and rust-analyzer ("library-ification"), which should eventually lead to rustc and rust-analyzer being two front-ends over a shared codebase.
```

Comment:
```suggestion
As detailed below, one major concern with rust-analyzer as it stands today is that it shares very little code with rustc. To avoid creating an unsustainable maintainance burden, this RFC proposes extracting shared libraries that will be used by both rustc and rust-analyzer ("library-ification"), which should eventually lead to rustc and rust-analyzer being two front-ends over a shared codebase.
```